File: sdk/v1.0.9/python/mediaviz_sdk/o_auth_clients.py
from __future__ import annotations
from typing import Any
from urllib.parse import quote, urlencode
import httpx

from .errors import handle_response


class OAuthClients:
    def __init__(self, ctx) -> None:
        self._ctx = ctx

    def create_client(
        self,
        clientName: str,
        clientType: str,
        redirectUris: list,
        isFirstParty: bool,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/oauth/clients'
        body = {k: v for k, v in {
            'client_name': clientName,
            'client_type': clientType,
            'redirect_uris': redirectUris,
            'is_first_party': isFirstParty,
        }.items() if v is not None}
        return self._ctx.client.request(path, 'POST', self._ctx.access_token, self._ctx.refresh_token, body).data

File: sdk/v1.0.24/javascript/oauth_clients.js
function stripUndef(o) { const r = {}; for (const k in o) if (o[k] !== undefined) r[k] = o[k]; return r; }

export class OauthClients {
  constructor(ctx) { this._ctx = ctx; }

  async createClient(clientName, clientType, redirectUris, isFirstParty) {
    this._ctx.requireTokens();
    const path = `/oauth/clients`;
    const body = stripUndef({
      client_name: clientName,
      client_type: clientType,
      redirect_uris: redirectUris,
      is_first_party: isFirstParty,
    });
    const { data } = await this._ctx.client.request(path, 'POST', this._ctx.accessToken, this._ctx.refreshToken, body);
    return data;
  }
}
